packages/toast/src/ControlledToast/ControlledToast.spec.tsx

```diff
@@ -70,6 +70,58 @@ describe('packages/toast/controlled', () => {
       expect(toast).toBeInTheDocument();
     });
 
+    test('does not render when `open` is true and component is unmounted', async () => {
+      const { queryByTestId } = render(
+        <ToastProvider>
+          {false && (
+            <Toast
+              open={true}
+              title="Test 1"
+              onClose={() => {}}
+              data-testid="test-toast1b"
+            />
+          )}
+        </ToastProvider>,
+      );
+
+      const toast = await waitFor(() => queryByTestId('test-toast1b'));
+      expect(toast).not.toBeInTheDocument();
+    });
+
+    test('unmounts when `open` is true and component render is toggled to false', async () => {
```

Comment:
@TheSonOfThomp I'm not sure if this is the best way to test this. To get it to work i had to change the data-testid for the rerendered Toast so I'm not sure if that is correct. Do you have any suggestions?

Reply:
I don't think in this case we're checking for the same component. I think after the `rerender` we want to 
```
await waitForElementToBeRemoved(toast)
expect(toast).not.toBeInTheDocument()
```